bundles/org.openhab.binding.energenie/src/main/java/org/openhab/binding/energenie/internal/handler/EnergeniePWMHandler.java

```diff
@@ -0,0 +1,132 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.energenie.internal.handler;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.energenie.internal.EnergeniePWMStateEnum;
+import org.openhab.binding.energenie.internal.config.EnergenieConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EnergeniePWMHandler} is responsible for reading states and update PWM channels.
+ *
+ * @author Hans-Jörg Merk - Initial contribution
+ */
+
+@NonNullByDefault
+public class EnergeniePWMHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(EnergeniePWMHandler.class);
+
+    private static final int HTTP_TIMEOUT_MILLISECONDS = 6000;
+
+    private String host = "";
+    private String password = "";
+    private int refreshInterval;
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    public EnergeniePWMHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            scheduler.execute(this::getState);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        EnergenieConfiguration config = getConfigAs(EnergenieConfiguration.class);
+
+        if (!config.host.isEmpty() && !config.password.isEmpty()) {
+            host = config.host;
+            password = config.password;
+            refreshInterval = EnergenieConfiguration.DEFAULT_REFRESH_INTERVAL;
+            logger.debug("Initializing EnergeniePWMHandler for Host '{}'", host);
+            updateStatus(ThingStatus.ONLINE);
+            onUpdate();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Can not access device , IP-Address or password not set");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("EnergeniePWMHandler disposed.");
+        final ScheduledFuture<?> refreshJob = this.refreshJob;
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
```

Comment:
```suggestion
        if (refreshJob != null) {
```

calling `cancel` again doesn't hurt and I think you want to set the class-field to `null` if it is non-null and the jopb cancelled

Reply:
Done